### PR TITLE
Python 3.8 test pipeline is not cancelled when build is cancelled

### PR DESCRIPTION
**Pull request by praveenkuttappan**
job condition is not checking for succeeded or failed status

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/identity/azure-identity/dev_requirements.txt [hunk 1]
```diff
@@ -2,3 +2,4 @@
 aiohttp;python_full_version>="3.5.2"
 typing_extensions>=3.7.2
 
+
```
Comment (chlowell) [inline sdk/identity/azure-identity/dev_requirements.txt]: Unintentional change?

Comment (praveenkuttappan) [reply sdk/identity/azure-identity/dev_requirements.txt]: Nope. This is to trigger CI for identity. i will not include this change when i merge.

**Pull request merged**